akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -542,6 +542,25 @@ trait FlowOps[+Out, +Mat] {
   def scan[T](zero: T)(f: (T, Out) ⇒ T): Repr[T, Mat] = andThen(Scan(zero, f.asInstanceOf[(Any, Any) ⇒ Any]))
 
   /**
+   * Similar to `scan` but only emits its result when the upstream completes,
+   * after which it also completes. Applies the given function towards its current and next value,
+   * yielding the next current value.
+   *
+   * If the function `f` throws an exception and the supervision decision is
+   * [[akka.stream.Supervision.Restart]] current value starts at `zero` again
+   * the stream will continue.
```

Comment:
Yes, this is configurable, you can set the supervision strategy stage by stage in the user API. My question is whether these stages (scan/fold) support the Resume strategy right now (maybe they do, but I don't remember)